snuba/admin/static/clickhouse_queries/index.tsx

```diff
@@ -1,20 +1,137 @@
-import React from "react";
+import React, { useState, useEffect } from "react";
 import Client from "../api_client";
 import { Table } from "../table";
 import QueryDisplay from "../components/query_display/query_display";
 import { QueryResult } from "../components/query_display/types";
 
+type PredefinedQuery = {
+  name?: string;
+  sql?: string;
+  description?: string;
+  selected?: boolean;
+};
+
 function ClickhouseQueries(props: { api: Client }) {
+  const [predefinedQuery, setPredefinedQuery] = useState<PredefinedQuery>({});
+  const [predefinedQueryOptions, setPredefinedQueryOptions] = useState<
+    PredefinedQuery[]
+  >([]);
+
+  useEffect(() => {
+    props.api.getPredefinedQueryOptions().then((res) => {
+      res.forEach(
+        (queryOption) => (queryOption.sql = formatSQL(queryOption.sql))
+      );
+      setPredefinedQueryOptions(res);
+    });
+  }, []);
+
   function tablePopulator(queryResult: QueryResult) {
     return (
       <Table headerData={queryResult.column_names} rowData={queryResult.rows} />
     );
   }
-  return QueryDisplay({
-    api: props.api,
-    endpoint: "run_clickhouse_system_query",
-    resultDataPopulator: tablePopulator,
-  });
+
+  function updatePredefinedQuery(queryName: string) {
+    const selectedQuery = predefinedQueryOptions.find(
+      (query) => query.name === queryName
+    );
+    setPredefinedQuery(() => {
+      return {
+        ...selectedQuery,
+        selected: false,
+      };
+    });
+  }
+
+  function selectPredefinedQuery() {
+    setPredefinedQuery((prevQuery) => {
+      return {
+        ...prevQuery,
+        selected: true,
+      };
+    });
+  }
+
+  function clearPredefinedQuery() {
+    setPredefinedQuery(() => {
+      return {};
+    });
+  }
+
+  function formatSQL(sql: string) {
+    const formatted = sql
+      .split("\n")
+      .map((line) => line.substring(4, line.length))
+      .join("\n");
+    return formatted.trim();
+  }
+
+  return (
+    <div>
+      <div>
+        <form>
+          <select
+            value={predefinedQuery.name || ""}
+            onChange={(evt) => updatePredefinedQuery(evt.target.value)}
+            style={selectStyle}
+          >
+            <option disabled value="">
+              Select a predefined query
+            </option>
+            {predefinedQueryOptions.map((option: PredefinedQuery) => (
+              <option key={option.name} value={option.name}>
+                {option.name}
+              </option>
+            ))}
+          </select>
+        </form>
+        {predefinedQuery?.sql && (
+          <div>
+            {
+              <div
+                style={{
+                  fontSize: 14,
+                  marginBottom: 5,
+                  width: "50vw",
+                }}
+              >
+                <p
+                  dangerouslySetInnerHTML={{
+                    __html: predefinedQuery.description || "",
+                  }}
```

Comment:
Hmm, this seems a bit odd. Was there a reason we need to resort to this way of setting the content of the element?

Reply:
I just wanted to be able to format the description with some html on the backend and just have it render here. This is def not ideal tho. Might be nice to render markdown here, and use something like https://www.npmjs.com/package/markdown-to-jsx, but i dk how strict we are being with adding extra dependencies

Reply:
I think we avoid sending HTML from the server to the client. Why can't we just send the SQL string?

Reply:
we do send the sql string, this is just for the description of the query

Reply:
This is because one of the descriptions has bold text? I think we can live without that.

Today, yes it is true that the possibility of say dodgy links or scripts being placed here may be small since we don't have a lot of canned queries and all of them have gone through Snuba code review process. But it's also very possible we might miss something in code review. And if allow users to edit and define these dynamically in the future it will open a can of worms.

I would prefer not to open the door to the possibility of HTML injection even if it seems unlikely today.